Ignore protected directories when counting objects

replaceafill opened the pull request:
This PR fixes object counting when access-protected directories exist in the transfer source. Under those circumstances `scandir.scandir` raises an `OSError` exception which is now handled and tested properly.

Connected to https://github.com/archivematica/Issues/issues/1019

sevein commented on storage_service/locations/tests/test_space.py:
```diff
@@ -0,0 +1,131 @@
+        },
+    }
+
+    # restrict read access to thrid directory (child of second)
```
https://nedbatchelder.com/blog/201401/comments_should_be_sentences.html

sevein reviewed (approved):
LGTM

replaceafill committed:
Ignore protected directories when counting objects

This fixes object counting when access-protected directories exist in
the transfer source. Under those circumstances `scandir.scandir` raises
an `OSError` exception which is now handled and tested properly.
